### ISSUE TITLE
Fix bed&breakfast calculation when stock is split

### ISSUE THREAD

**Issue by sygi** [KapJI/capital-gains-calculator#511]
The stock split action in the code is considered a 0-priced acquisition. This works as long as there is no bed&breakfast to consider (no sale in the previous 30 days).

It would be nice to fix the calculation to handle this case as well, or, at a very least, to throw an error to avoid silently incorrect calculation.

**Comment by vmartinv** [comment]
Can you add some minimal examples to test in whatever csv format you are using? Alongside what should be correct reported gain/loss, these cases would help:
- Buy, sell, split, buy within 30 days
- Buy, sell, split within 30 days
- Buy, sell, split, buy after 30 days

**Comment by sygi** [comment]
I attach a CSV file with an example split. I also created a test here: #541 which shows where the problem lies (the correct CG is 7. whereas the library calculates 16., as it considers split as an 0-cost acquisition).
[schwab.csv](https://github.com/user-attachments/files/16066608/schwab.csv)

**Comment by vmartinv** [comment]
> I attach a CSV file with an example split. I also created a test here: #541 which shows where the problem lies (the library consider a split as an 0-cost acquisition). [schwab.csv](https://github.com/user-attachments/files/16066608/schwab.csv)

I wouldn't worry about fixing it at the moment, I had one idea about it. Just having the 3 examples I listed would be helpful

**Comment by sygi** [comment]
Please find the three examples attached. Note that no2 and 3 are actually the same in terms of cgt, as both split and buy after 30 days are tax neutral.

[schwab2_5_00.csv](https://github.com/user-attachments/files/16081525/schwab2_5_00.csv)
[schwab_5_00.csv](https://github.com/user-attachments/files/16081527/schwab_5_00.csv)
[schwab_18_00.csv](https://github.com/user-attachments/files/16081528/schwab_18_00.csv)

**Comment by KapJI** [comment]
I guess the first thing we should do is detect b&b when split is involved and abort as unsupported case. Until someone figure out how to calculate this cases properly.

**Comment by alexpung** [comment]
I am quite sure you do the b&b matching rules with the matching quantity adjusted for the share split ratio.
https://www.gov.uk/government/publications/share-reorganisations-company-takeovers-and-capital-gains-tax-hs285-self-assessment-helpsheet/hs285-share-reorganisations-company-takeovers-and-capital-gains-tax-2021

**Comment by sygi** [comment]
Yes, this is what I advocate above and propose strategies for implementing io #541
